client/src/components/Restaurants.js

```diff
@@ -0,0 +1,37 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+function Restaurants(props) {
+  //Setting Beginning State for Restaurant Names
+  const [restNames, setRestNames] = useState([]);
+
+  //Fetch for Restaurant Names
+  useEffect(() => {
+    //Guard clause for fetch
+    if (restNames.length === 0) {
+      fetch("apis")
+        .then((res) => res.json())
+        .then((restList) => {
+          setRestNames(restList);
+        });
+    }
+  });
+
+  return (
+    <div>
+      <ul>
+        {/* Iterating Restaurant Names */}
+        {restNames.map((id, index) => {
+          return (
+            <h2 key="index">
+              {/* Presenting Name Iterations */}
+              <Link to={`/api/${id}`}>{id}</Link>
+            </h2>
+          );
+        })}
```

Comment:
I noticed a lot of the warnings on the browser console are about a key. I then saw you had tried to resolve that by giving each `h2` it's own key of the index #. The only reason this isn't currently working is a slight syntax mistake. Changing line 26 to `<h2 key={index}>` with the {} brackets instead of "" should resolve this.